homeassistant/components/fronius/__init__.py

```diff
@@ -28,24 +31,58 @@
 )
 
 _LOGGER = logging.getLogger(__name__)
+PLATFORMS: list[str] = ["sensor"]
 
 FroniusCoordinatorType = TypeVar("FroniusCoordinatorType", bound=FroniusCoordinatorBase)
 
 
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up fronius from a config entry."""
+    host = entry.data[CONF_HOST]
+    fronius = Fronius(async_get_clientsession(hass), host)
+    solar_net = FroniusSolarNet(hass, entry, fronius)
+    await solar_net.init_devices()
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = solar_net
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+    # reload on config_entry update
+    entry.async_on_unload(entry.add_update_listener(async_update_entry))
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    if unload_ok:
+        solar_net = hass.data[DOMAIN].pop(entry.entry_id)
+        while solar_net.cleanup_callbacks:
+            solar_net.cleanup_callbacks.pop()()
+
+    return unload_ok
+
+
+async def async_update_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
```

Comment:
This function shouldn't return anything. Nothing is checking its return value.

Reply:
Thank you! Suggestions are applied in #60094